### ISSUE TITLE
vue tag pro

### ISSUE THREAD

**Issue by ysyfff** [vincaslt/vscode-highlight-matching-tag#20]
It will not works on this kind of tag like `<div :xx="aa"></div>`

**Comment by kjakobson** [comment]
Got my .vue files working with this regex: `<\S+?(\s((\{.*?\})|((\w|-|\.|(:\w)|(@\w))+(\s?|(=((\'[^\']*?\')|(\{[^\}]*?\})|(\"[^\"]*?\")))))\s*)?)*?\/?>`.  You can check it here: https://regex101.com/r/E0tnAy/4.

I had to change it in extension source file (%USERPROFILE%\.vscode\extensions). Not sure if this doesn't brake any other syntaxes. Maybe it can be shipped in some future release, what do you think @vincaslt ?

**Comment by thedonquixotic** [comment]
I'm having a similar problem with my Angular html template files not matching once I use Angular specific features such as [square brackets] for property binding. 

I've got a little experience lately doing regex stuff for the Hilighter extension (it's like todo highlight but more full featured regex use), but it sounds like you're saying, @kjakobson  you have to actually go in and mod the extension file to get it to work? That seems like a not great solution once the extension updates. Though it doesn't seem to be under active development :/ 

I'm going to try using your solution with my Hilighter extension and see if that does basically what I need. If I learn enough regex I imagine I can make something like hilights matching tags extension obsolete. It's nice having simple settings to manipulate in my settings.json but it seems to have pretty limited use outside of pure html stuff.

**Comment by thedonquixotic** [comment]
Ooookay. Looking at the regex and I see why there's an advantage to this extension, because while the Hilighter extension can use regex to find the tag, it won't find the matching tag. And also this extension can intelligently find the matching closing tag, which I imagine is more involved than just regex. 


Would it be possible to add regex option to Hilight Matching Tags similar to the Hilight extension (https://github.com/fabiospampinato/vscode-highlight) @vincaslt   ?   That way when people use a framework where the tags don't work or create a custom tag of some kind, they could fix it themselves. It could mean that adding multiple types of styles for multiple types of tags would be possible which would be super useful!

**Comment by vincaslt** [comment]
Should be fixed with v0.5, test cases cover it now.

I will definitely consider adding custom tag definitions after this extension has stabilized (there were many problems with the algorithm). Now there's actual possibility of extending it.

**Comment by jcklpe** [comment]
Just for future reference in case any person needed the info, the regex I ended up using to highlight HTML tags using the Highlighter app is as follows:

```
//- Opening  HTML Tag
        "(<[^\\/!%=\\?])([^!%\\?(</)](.|\n|\r)*?)(>)": [
          {
            "borderWidth": "1px 0 0 4px",
            "borderStyle": "solid",
            "borderColor": "rgba(255, 103, 123, 0.5)",
            "borderRadius": "0px",
            "backgroundColor": "rgba(33, 24, 28, 0.6)",
            "color": "#d44040",
            "fontStyle": "bold",
            "fontWeight": "bold",
          },

          {
            "backgroundColor": "rgba(26, 18, 26, 0.4)",
            "borderWidth": "0 0 0 0",
            "borderStyle": "solid",
            "borderColor": "rgba(255, 103, 123, 0.5)",
            "borderRadius": "0px",
          },

          {
            "backgroundColor": "rgba(15, 21, 31, 0.45)",
            "color": "rgba(40, 68, 85, 0.7)",
            "fontStyle": "bold",
            "fontWeight": "bold"
          },

        ],
        //- Closing HTML Tag
        "(<\\/)((.|\n|\r)*?)((?!\\?)>)": [
          {
            "backgroundColor": "rgba(15, 21, 31, 0.45)",
            "color": "rgba(40, 68, 85, 0.7)"
          },

          { "borderWidth": "0 0 0 0",
            "borderStyle": "solid",
            "borderColor": "rgba(255, 103, 123, 0.5)",
            "borderRadius": "0px",
            "backgroundColor": "rgba(26, 18, 26, 0.4)",

          },

          {
            "borderWidth": "0 4px 1px 0",
            "borderStyle": "solid",
            "borderColor": "rgba(255, 103, 123, 0.5)",
            "borderRadius": "0px",
            "backgroundColor": "rgba(33, 24, 28, 0.6)",
            "color": "#d44040",
            "fontStyle": "bold",
            "fontWeight": "bold",
          },
        ],

        // PHP Tag
        "(<\\?php)((.|\n|\r)*?)(\\?>)": [
          {
            "borderWidth": "1px 0 0 4px",
            "borderStyle": "solid",
            "borderColor": "#9075ac",
            "borderRadius": "0px",
            "backgroundColor": "rgba(33, 24, 28, 0.6)",
            "color": "#9075ac",
            "fontStyle": "bold",
            "fontWeight": "bold",
          },

          {
            "backgroundColor": "rgba(26, 18, 26, 0.4)",
            "borderWidth": "0px 0 0 0",
            "borderStyle": "solid",
            "borderColor": "#9075ac",
            "borderRadius": "0px",
          },

          {
            "borderWidth": "0 4px 1px 0",
            "borderStyle": "solid",
            "borderColor": "#9075ac",
            "borderRadius": "0px",
            "backgroundColor": "rgba(15, 21, 31, 0.45)",
            "color": "#9075ac"
          },
        ],

```

I ended up separating my regex rules into opening and closing tags because trying to create a nested group regex that could reliably pick ALL opening tags, their interior contents, and the proper paired closing tag was basically impossible (which I think has generally been acknowledged as the trouble with regexing HTML due to it's nested nature). 

IDK how the matching tags regex works but I imagine it's rather clever and has to trade it's ability to match the correct pair of tags for my modular regex search approach which is capable of getting pretty much any html tag no matter what attributes, line breaks, or framework specific features are used, but comes at the cost of being able to actually match tags. If there can be some way to combine the two approaches in an advantageous way, or as suggested perhaps provide an easy to use format for creating exceptional cases, that would be really cool and helpful :) I'm going to go look at whatever merge notes show these changes as I'm pretty curious how you made the changes! I'm very much a noob at algo stuff.
